File: algorithms/23_elixir/dp/lcs.ex
defmodule LCS do
  def lcs(a, b) when is_binary(a) and is_binary(b) do
    lcs(String.graphemes(a), String.graphemes(b))
  end
  def lcs([], _), do: 0
  def lcs(_, []), do: 0
  def lcs([h|t1], [h|t2]), do: 1 + lcs(t1, t2)
  def lcs([_|t1]=a, [_|t2]=b), do: max(lcs(a, t2), lcs(t1, b))
end

File: algorithms/03_cpp/dp/lcs.cpp
#include <string>
#include <vector>
#include <algorithm>
int lcs(const std::string& a, const std::string& b){
    int m=a.size(),n=b.size(); std::vector<std::vector<int>> dp(m+1,std::vector<int>(n+1,0));
    for(int i=1;i<=m;i++) for(int j=1;j<=n;j++)
        dp[i][j]=a[i-1]==b[j-1]?dp[i-1][j-1]+1:std::max(dp[i-1][j],dp[i][j-1]);
    return dp[m][n];
}